### DISCUSSION TITLE
A question reagrding partial upload

### DISCUSSION THREAD

**Discussion by omar391** [cloudflare/wrangler-action#244] [Q&A]
I know Cloudflare Wrangler upload only changed files. But isn't it problematic, since the static site builder bundle and name files by hash ie. myfile.ajd29357.js are unique and get a new name on every change? So CF will keep myfile.ajd29357.js  even though it is old and will never be updated again.

Regards.

**Comment by ferferga** [comment]
The hash is generated based on the contents of the file, so if the outputted files changed, the hash will change as well.

**Reply by omar391** [reply]
No, I understand that. My question was whether we keep old hashed files or not. If we do, then it would exceed 20k files limit much earlier.

**Reply by ferferga** [reply]
No idea that there was a limit imposed by Cloudflare. I assume that the oldest entries are purged (like GitHub Action's caches and any other LRU implementation).
